### ISSUE TITLE
Legend overlaps x axis ticks on negative tick rotation angle

### ISSUE THREAD

**Issue by Soly** [c3js/c3#1298]
Paste the following code into http://c3js.org/samples/axes_x_tick_rotate.html

``` javascript
datum =  {"axes":{"data1":"y","data2":"y2","data3":"y","data4":"y2"},"cols":[["x","2005-07-01","2005-06-01","2005-05-01","2005-04-01","2005-03-01","2005-02-01","2005-01-01","2004-12-01","2004-11-01","2004-10-01","2004-09-01"],["data1","79.6","39.1","86.7","68.6","49.2","69.1","59.1","29.1","17.4","56.2","87.5"],["data2","63","40","46","58","27","24","82","191","555","66","772"],["data3","9.9","8.7","18.7","48.5","58.7","58.2","68.7","78.3","88.2","98.7","100"],["data4","10","10","30","50","0","30","20","70","50","20","10"]],"types":{"data1":"line","data3":"line","data2":"bar","data4":"bar"}}
var chart = c3.generate({
      data: {
        x: 'x',
        columns: datum.cols,
        types: datum.types,
        axes: datum.axes
      },
      axis: {
        x: {
          type: 'timeseries',
          tick: {
            format: '%b %Y',
            culling: false,
            rotate: -20
          }
        },
        y: {
          label: {
            text: 'Y1',
            position: 'outer-middle'
          },
          tick: {
            format: function(d) { return d + '%'; }
          },
        },
        y2: {
          label: {
            text: 'Y2',
            position: 'outer-middle'
          },
          show: true,
        }
      }
});
```

![image](https://cloud.githubusercontent.com/assets/3904838/8966097/60d5e6dc-35fd-11e5-8ee3-b90d8cdad1ea.png)

(Everything does adjust appropriately for positive rotation angles. Just not negative)

**Comment by roldan** [comment]
Same problem. Have you found a workaround @Soly ?

**Comment by Soly** [comment]
@roldan I think I ended up changing the [legend position](http://c3js.org/reference.html#legend-position) to 'right'. It wasn't exactly optimal as my graph was nested in a smaller panel, but it worked.

**Comment by GoekhanOezen** [comment]
hi look at #1511

**Comment by alexbain** [comment]
I opened a PR with a fix for this, thanks to direction from #1511 https://github.com/c3js/c3/pull/1766

**Comment by panthony** [comment]
Closing in favor of #1511
